EFCore2Playgrounds/Program.cs

```diff
@@ -95,22 +93,23 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
 
         protected override void OnModelCreating(ModelBuilder modelBuilder)
         {
-            modelBuilder.Entity<State>()
-                .Property(x => x.Id).ValueGeneratedNever();
+            modelBuilder.Entity<State>() .Property(x => x.Id).ValueGeneratedNever();
             modelBuilder.Entity<Order>()
                 .HasOne(typeof(State), nameof(State))
                 .WithMany()
                 .HasForeignKey("StateId");
+
             modelBuilder.Entity<OrderCategory>()
-                .HasKey(x => new {x.OrderId, x.CategoryId});
+                .HasKey(x => new { x.OrderId, x.CategoryId });
+
             modelBuilder.Entity<OrderCategory>()
-                .HasOne(typeof(Order), nameof(Order))
-                .WithMany()
+                .HasOne(oc=>oc.Order)
+                .WithMany("_orderCategories")
```

Comment:
tell EF that the private property of name "_orderCategories" maps to this navigation property